downloads/swan-lake-release-notes/swan-lake-2201.7.0/RELEASE_NOTE.md

```diff
@@ -0,0 +1,232 @@
+
+## Language updates
+
+### New features
+
+#### Introduction of the `group by` and `collect` clauses
+
+The language now supports the `group by` and `collect` clauses to perform aggregation-related operations. The `group by` clause is used to group a collection based on a `grouping-key`. The `grouping-key` will be unique for each group. The `collect` clause is used to group a collection into one group. 
+
+```ballerina
+import ballerina/io;
+public type CovidEntry record {|
+    string district;
+    string province;
+    decimal deaths;
+|};
+public final table<CovidEntry> covidTable = table [
+    {district: "Colombo", province: "Western", deaths: 21},
+    {district: "Kandy", province: "Central", deaths: 14},
+    {district: "Kaluthara", province: "Western", deaths: 18},
+    {district: "Jaffna", province: "North", deaths: 10}
+];
+public function main() {
+    record {|string province; decimal deaths;|}[] deathsByProvince = from var {province, deaths} in covidTable
+        group by province
+        select {province, deaths: sum(deaths)};
+    io:println(deathsByProvince);
+    decimal totalDeaths = from var {deaths} in covidTable
+            collect sum(deaths);
+    io:println(totalDeaths);
+}
+```
+
+### Improvements
+
+#### Support for constant declarations without a type descriptor
+
+Constants can now be declared without specifying a type descriptor. Previously, it was required to specify a type descriptor except with numeric and string literal expressions.
+
+Both of the following cases are allowed now.
+
+```ballerina
+const map<string> HTTP_OK = {httpCode: "200", message: "OK"};
+const HTTP_BAD_REQUEST = {httpCode: "400", message: "BAD REQUEST"};
+```
+
+#### Support for unions as type descriptors in constant declarations
+  
+Constant declarations with unions as type descriptors are now supported.
+
+```ballerina
+const float|decimal TOTAL = 1 + 2.0;
+```
+
+#### Support for more recursive type definitions
+
+The type definition resolution logic has been improved to support more recursive type definitions.
+
+```ballerina
+type A [A];
+type B [B] & readonly;
+```
+
+### Bug fixes
+
+To view other bug fixes, see the [GitHub milestone for Swan Lake 2201.7.0](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+label%3ATeam%2FCompilerFE+milestone%3A2201.7.0+is%3Aclosed+label%3AType%2FBug).
+
+## Runtime updates
+
+### Bug fixes
+
+To view bug fixes, see the [GitHub milestone for 2201.7.0 (Swan Lake)](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.7.0+label%3ATeam%2FjBallerina+label%3AType%2FBug+is%3Aclosed).
+
+## Standard library updates
+
+### New features
+
+#### `http` package
+
+- Introduced the `httpscerr` module, which contains the HTTP status code errors.
```

Comment:
Do we need to include this to release note?

Reply:
No, will remove it.

Reply:
Removed via [89a9dda](https://github.com/ballerina-platform/ballerina-dev-website/pull/7235/commits/89a9dda02f126b50dcc8ea8a5774b7f215bd7905)